test/jdk/jdk/jfr/jvm/TestGetEventWriter.java

```diff
@@ -51,6 +57,9 @@
  *
  * @run main/othervm jdk.jfr.jvm.TestGetEventWriter
  *
+ * @run main/othervm -XX:+UnlockExperimentalVMOptions -XX:+EnableJVMCI -Dtest.jvmci=true --add-exports=jdk.jfr/jdk.jfr.internal.event=ALL-UNNAMED
```

Comment:
I think `--add-exports=jdk.jfr/jdk.jfr.internal.event=ALL-UNNAMED` should be added to all `@run` specs in this test. See more details [here](https://github.com/openjdk/jdk/pull/8383#discussion_r915755771).

Reply:
Maybe I'm missing the point of these tests but I assume they are trying to exercise the blessed method checks in which case the IllegalAccessError message should be coming from here: https://github.com/openjdk/jdk/blob/master/src/hotspot/share/jfr/instrumentation/jfrResolution.cpp#L81
Instead, it is coming from the module access checks.
Please see more details at https://github.com/openjdk/jdk/pull/8383#discussion_r915755771

Reply:
Now I understand better what you are referring to, thanks for noticing this. I think your suggestion is good, it makes sense to make the tests better targeted in that sense. Please also add the --add-exports to the other tests as well. Thanks again for seeing this.

Reply:
I think we want to test what is being shipped. From the beginning there was --add-exports, but then it was removed and we found issues (unexpected exception). Maybe we can have both, but I think it would be unfortunate to not test the "full scenario"

Reply:
As far as I can see, without the `--add-exports`, all that is being tested is module access checks. Are there other tests specifically testing the access checks in [`JfrResolution::on_runtime_resolution`](https://github.com/openjdk/jdk/blob/aa9b8f04bf74d5fa00f2b27895e7369abea3a930/src/hotspot/share/jfr/instrumentation/jfrResolution.cpp#L55).
That said, I'm happy to leave the original tests in `TestGetEventWriter.java` as is.

Reply:
Ok Douglas, please leave the original tests as is. There are a few paths involved, and we need to remember why we ended up stating the tests as we did. We can revisit this later, so let's not block you from getting this in.